### PR TITLE
Automate docs version list update PR

### PR DESCRIPTION
**Pull request by iethree**
Trying a different approach to https://github.com/metabase/metabase/pull/37613

Replaces version list update babashka script with a couple lines of javascript in the release github action that opens a PR with the currently-releasing version added to the list of releases in docs/releases.md

This saves a manual step for @jeff-bruemmer for every release. See e.g. https://github.com/metabase/metabase/pull/45316

sample run in my fork: 
- action run: https://github.com/iethree/metabase/actions/runs/9879020800
- pr: https://github.com/iethree/metabase/pull/26

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -511,22 +511,72 @@ jobs:
...
+      contents: write
+      pull-requests: write
+    env:
+      NEW_OSS_VERSION: ${{ needs.check-version.outputs.oss }}
+      NEW_EE_VERSION: ${{ needs.check-version.outputs.ee }}
+      GH_TOKEN: ${{ secrets.METABASE_AUTOMATION_USER_TOKEN }}
     steps:
       - uses: actions/github-script@v7
         with:
-          github-token: ${{ secrets.DOCS_UPDATE_TOKEN }}
```
Comment (iethree) [inline .github/workflows/release.yml:520]: we should delete this old token

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -511,22 +511,72 @@ jobs:
...
+          git commit -m "Update release version list to include $NEW_OSS_VERSION"
+          git push origin update-version-list-$NEW_OSS_VERSION
+
+      - name: open pr with updates
+        run: | # bash
+          gh pr create --fill \
+            --label "Type:Documentation" \
+            --label "backport" \
+            --reviewer "jeff-bruemmer" \
+            --assignee "jeff-bruemmer"
```
Comment (iethree) [inline .github/workflows/release.yml:579]: should we have a writers github group?

Comment (jeff-bruemmer) [reply .github/workflows/release.yml:580]: Probably, but let's worry about that in a future PR.

**Review by jeff-bruemmer — APPROVED** [review]
Approving to unblock:

As discussed:

- Let's keep the old script as a backup for now. We can always remove it later.
- This code won't cover older point releases, so we'll have to manually reorder. We can automate that later if it's a pain.
- The link to the release will trigger link errors until the release is published; just something to be aware of.

Thanks @iethree! This is great. Will definitely save some hassle.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -511,22 +511,73 @@ jobs:
 
   trigger-docs-update:
     if: ${{ inputs.skip-release-notes != true }}
-    needs: push-tags
+    needs:
+      - push-tags
```
Comment (uladzimirdev) [inline .github/workflows/release.yml:515]: lets maybe use array like [] https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -511,22 +511,73 @@ jobs:
...
+      GH_TOKEN: ${{ secrets.METABASE_AUTOMATION_USER_TOKEN }}
     steps:
       - uses: actions/github-script@v7
         with:
-          github-token: ${{ secrets.DOCS_UPDATE_TOKEN }}
-          script: |
-            github.rest.repos.createDispatchEvent({
+          github-token: ${{ secrets.METABASE_AUTOMATION_USER_TOKEN }}
+          script: | # js
+            await github.rest.repos.createDispatchEvent({
```
Comment (uladzimirdev) [inline .github/workflows/release.yml:531]: Curious if top level await works

**Review by uladzimirdev — APPROVED** [review]

File: .github/workflows/release.yml (hunk 4 above)
Comment (iethree) [reply .github/workflows/release.yml:531]: yep! actions script runs as [the body of an async function](https://github.com/actions/github-script#:~:text=the%20body%20of%20an%20asynchronous%20function%20call)

File: .github/workflows/release.yml (hunk 1 above)
Comment (nemanjaglumac) [reply .github/workflows/release.yml:520]: Deleted.

**Pull request merged**